src/recipe_dictionary.cpp

```diff
@@ -121,6 +121,13 @@ std::vector<const recipe *> recipe_subset::search( const std::string &txt,
     return res;
 }
 
+bool recipe_subset::empty_category( const std::string &cat,
+                                    const std::string &subcat ) const
+{
+    // @todo: Make more efficient
+    return in_category( cat, subcat ).size() == 0;
```

Comment:
why not just ```.empty()```?

Reply:
If it is the todo comment you are referring what I meant it than in_category builds a vector that are not really needed however I was tired a wasn't in the mood to copy and adapt the code of in_category.

If you are referring to that .empty() for vectors is marginally more efficient that .size() == 0, then you are right as well. C++ is not my main language, I prefer Python, so I do not always select the most optimal way to do it when programming in C++. In addition since most of what I do professionally has database access as the dominant source of slowness. I usually care more about readability than speed.

In any case, since you have mentioned it, I will try to implement what I meant by the todo comment, later. However, since needlessly calculating and sizing a small vector is very fast anyway, I don't consider it a priority.

Premature optimization is the root to all evil. :-)

In short, noted, but I don't consider it a show stopper for merging.

Reply:
@illi-kun I have made the method you complained about more efficient now and resubmitted it rebase onto the latest master. Especially see commit 15b6313. This others are the same, I just split them in separate commits.

Reply:
Perhaps I should mention that I experimented with making the disabled subtabs italic in addition to dark gray. However, I only got it to work on the non-tiles version so I put i on ice. I will revisit the issue at a later time. Until then dark gray will have to suffice.